### DISCUSSION TITLE
Can Maddy compile for Windows?

### DISCUSSION THREAD

**Discussion by lonnietc** [emersion/maddy#514] [Q&A]
Hello,

I'm looking for a reasonably complete Mail server that can compile cross platform and is written in Golang so that it could potentially be run locally.

Can Maddy be compiled for Windows?
Thanks

**Comment by foxcpp** [comment]
Yes it can.

Some considerations:
1. You need MinGW toolchain to compile SQLite3 support, which is required for default config.
2. build.sh is not usable, you will need to run `go build ./cmd/maddy` directly.
3. Default configuration path is hardcoded to /etc/maddy/maddy.conf, you will need to always pass -config option when running the server.
4. Default state_dir/runtime_dir values are also POSIX-specific, you will need to always set these in the config to appropriate values.
